### PR TITLE
fix: plugin sort error

### PR DESCRIPTION
**Pull request by clChenLiang**
# PR Details

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] My code follows the code style of this project.
- [ ] I have updated changeset
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: packages/toolkit/utils/src/dagSort.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    });
+  });
+
+  // search the zero input plugin
+  let zeroEndPoints = plugins.filter(
+    item => !allLines.find(l => l[1] === item[key]),
+  );
+
+  const sortedPoint: P[] = [];
+  // 有环的处理
```
Comment (xuchaobei) [inline packages/toolkit/utils/src/dagSort.ts:40]: in English?

File: packages/toolkit/utils/src/dagSort.ts [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  while (zeroEndPoints.length) {
+    const zep = zeroEndPoints.shift();
+    sortedPoint.push(getPluginByAny(zep!));
+    //   zeroEndPoints.shift();
```
Comment (xuchaobei) [inline packages/toolkit/utils/src/dagSort.ts:44]: 多余代码？

**Review by xuchaobei — CHANGES_REQUESTED** [review]

File: packages/toolkit/plugin/src/manager/shared.ts [hunk 3]
```diff
@@ -1,3 +1,4 @@
+import { dagSort } from '@modern-js/utils';
```
Comment (chenjiahan) [inline packages/toolkit/plugin/src/manager/shared.ts:1]: `@modern-js/plugin` 是一个 unversal 的包，运行时代码也会引入的，这个包不能引用 `@modern-js/utils`

File: packages/toolkit/utils/src/dagSort.ts (hunk 1 above)
Comment (clChenLiang) [reply packages/toolkit/utils/src/dagSort.ts]: done

File: packages/toolkit/plugin/src/manager/shared.ts [hunk 4]
```diff
@@ -1,3 +1,4 @@
+import { dagSort } from '../../../utils/src/dagSort';
```
Comment (chenjiahan) [inline packages/toolkit/plugin/src/manager/shared.ts:1]: 不能跨包相对路径引用

Comment (clChenLiang) [reply packages/toolkit/plugin/src/manager/shared.ts]: 移过来了。先给 toolkit/plugin 里加了一个 tool 的目录

**Review by xuchaobei — APPROVED** [review]

**Review by chenjiahan — APPROVED** [review]

**Review by yimingjfe — APPROVED** [review]

**Pull request merged**